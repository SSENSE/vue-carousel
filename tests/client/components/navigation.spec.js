import { mount } from '@vue/test-utils';

const Carousel = require('../../../src/Carousel.vue');
const Slide = require('../../../src/Slide.vue');

describe('Navigation', () => {
  let wrapper;
  let vm;
  let $navigation;

  beforeEach(() => {
    wrapper = mount(Carousel, {
      propsData: {
        navigationEnabled: true,
        perPage: 1
      },
      slots: {
        default: [Slide, Slide]
      }
    });
    vm = wrapper.vm;
    $navigation = wrapper.find('.navigation');
  });

  it('should mount successfully', done => {
    expect($navigation).toBeDefined();

    vm.$nextTick(() => {
      expect(wrapper).toMatchSnapshot();
      done();
    });
  });

  it('should render a next button', done => {
    expect(wrapper.find('.VueCarousel-navigation-next')).toBeDefined();

    vm.$nextTick(() => {
      expect(wrapper).toMatchSnapshot();
      done();
    });
  });

  it('should render a prev button', done => {
    expect(wrapper.find('.VueCarousel-navigation-prev')).toBeDefined();

    vm.$nextTick(() => {
      expect(wrapper).toMatchSnapshot();
      done();
    });
  });

  it('should trigger page advance when next is clicked', done => {
    vm.$nextTick(() => {
      expect(vm.currentPage).toBe(1);

      expect(wrapper).toMatchSnapshot();

      done();
    });
  });

  it('should trigger page advance backward when prev is clicked', done => {
    vm.goToPage(2);
    wrapper.find('.VueCarousel-navigation-prev').trigger('click');

    vm.$nextTick(() => {
      expect(vm.currentPage).toBe(1);

      expect(wrapper).toMatchSnapshot();

      done();
    });
  });
});
