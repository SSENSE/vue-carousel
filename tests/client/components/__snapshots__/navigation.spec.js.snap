exports[`Navigation should mount successfully 1`] = `
"div
  .VueCarousel-inner(style=\'visibility: hidden;\')
    .VueCarousel-slide
    .VueCarousel-slide
|  
.VueCarousel-pagination
  .VueCarousel-dot-container
    .VueCarousel-dot.VueCarousel-dot--active(style=\'margin-top: 20px; padding: 10px;\')
      .VueCarousel-dot-inner(style=\'width: 10px; height: 10px; background: rgb(0, 0, 0);\')
    .VueCarousel-dot(style=\'margin-top: 20px; padding: 10px;\')
      .VueCarousel-dot-inner(style=\'width: 10px; height: 10px; background: rgb(239, 239, 239);\')
|  
.VueCarousel-navigation
  a.VueCarousel-navigation-button.VueCarousel-navigation-prev.VueCarousel-navigation--disabled(href=\'#\', style=\'padding: 8px; margin-right: -8px;\') &#x25C0;
  |  
  a.VueCarousel-navigation-button.VueCarousel-navigation-next(href=\'#\', style=\'padding: 8px; margin-left: -8px;\') &#x25B6;
"
`;

exports[`Navigation should render a next button 1`] = `
"div
  .VueCarousel-inner(style=\'visibility: hidden;\')
    .VueCarousel-slide
    .VueCarousel-slide
|  
.VueCarousel-pagination
  .VueCarousel-dot-container
    .VueCarousel-dot.VueCarousel-dot--active(style=\'margin-top: 20px; padding: 10px;\')
      .VueCarousel-dot-inner(style=\'width: 10px; height: 10px; background: rgb(0, 0, 0);\')
    .VueCarousel-dot(style=\'margin-top: 20px; padding: 10px;\')
      .VueCarousel-dot-inner(style=\'width: 10px; height: 10px; background: rgb(239, 239, 239);\')
|  
.VueCarousel-navigation
  a.VueCarousel-navigation-button.VueCarousel-navigation-prev.VueCarousel-navigation--disabled(href=\'#\', style=\'padding: 8px; margin-right: -8px;\') &#x25C0;
  |  
  a.VueCarousel-navigation-button.VueCarousel-navigation-next(href=\'#\', style=\'padding: 8px; margin-left: -8px;\') &#x25B6;
"
`;

exports[`Navigation should render a prev button 1`] = `
"div
  .VueCarousel-inner(style=\'visibility: hidden;\')
    .VueCarousel-slide
    .VueCarousel-slide
|  
.VueCarousel-pagination
  .VueCarousel-dot-container
    .VueCarousel-dot.VueCarousel-dot--active(style=\'margin-top: 20px; padding: 10px;\')
      .VueCarousel-dot-inner(style=\'width: 10px; height: 10px; background: rgb(0, 0, 0);\')
    .VueCarousel-dot(style=\'margin-top: 20px; padding: 10px;\')
      .VueCarousel-dot-inner(style=\'width: 10px; height: 10px; background: rgb(239, 239, 239);\')
|  
.VueCarousel-navigation
  a.VueCarousel-navigation-button.VueCarousel-navigation-prev.VueCarousel-navigation--disabled(href=\'#\', style=\'padding: 8px; margin-right: -8px;\') &#x25C0;
  |  
  a.VueCarousel-navigation-button.VueCarousel-navigation-next(href=\'#\', style=\'padding: 8px; margin-left: -8px;\') &#x25B6;
"
`;

exports[`Navigation should trigger page advance backward when prev is clicked 1`] = `
"div
  .VueCarousel-inner(style=\'visibility: hidden;\')
    .VueCarousel-slide
    .VueCarousel-slide
|  
.VueCarousel-pagination
  .VueCarousel-dot-container
    .VueCarousel-dot(style=\'margin-top: 20px; padding: 10px;\')
      .VueCarousel-dot-inner(style=\'width: 10px; height: 10px; background: rgb(239, 239, 239);\')
    .VueCarousel-dot.VueCarousel-dot--active(style=\'margin-top: 20px; padding: 10px;\')
      .VueCarousel-dot-inner(style=\'width: 10px; height: 10px; background: rgb(0, 0, 0);\')
|  
.VueCarousel-navigation
  a.VueCarousel-navigation-button.VueCarousel-navigation-prev(href=\'#\', style=\'padding: 8px; margin-right: -8px;\') &#x25C0;
  |  
  a.VueCarousel-navigation-button.VueCarousel-navigation-next.VueCarousel-navigation--disabled(href=\'#\', style=\'padding: 8px; margin-left: -8px;\') &#x25B6;
"
`;

exports[`Navigation should trigger page advance when next is clicked 1`] = `
"div
  .VueCarousel-inner(style=\'visibility: hidden;\')
    .VueCarousel-slide
    .VueCarousel-slide
|  
.VueCarousel-pagination
  .VueCarousel-dot-container
    .VueCarousel-dot(style=\'margin-top: 20px; padding: 10px;\')
      .VueCarousel-dot-inner(style=\'width: 10px; height: 10px; background: rgb(239, 239, 239);\')
    .VueCarousel-dot.VueCarousel-dot--active(style=\'margin-top: 20px; padding: 10px;\')
      .VueCarousel-dot-inner(style=\'width: 10px; height: 10px; background: rgb(0, 0, 0);\')
|  
.VueCarousel-navigation
  a.VueCarousel-navigation-button.VueCarousel-navigation-prev(href=\'#\', style=\'padding: 8px; margin-right: -8px;\') &#x25C0;
  |  
  a.VueCarousel-navigation-button.VueCarousel-navigation-next.VueCarousel-navigation--disabled(href=\'#\', style=\'padding: 8px; margin-left: -8px;\') &#x25B6;
"
`;
